### ISSUE TITLE
Option to constraint generation to contain specific substructures

### ISSUE THREAD

**Issue by chaileon** [MehmetAzizYirik/MAYGEN#76]
Thanks for the great repo!

Would you be adding the functionality of constraining the generation to contain specific substructures? A simple way is to generate all possibilities, and discard those that do not contain the substructure.  However, a more efficient way would be to restrict the generation so that no time is wasted in generating invalid answers.

One approach could be to initialize the adjacency matrix with the substructure before generation. However, your algorithm first distributes hydrogens to atoms, which does not allow us to do that easily.

If you have any plans or thoughts in this direction, would be great if you can share them.

Thanks!

**Comment by MehmetAzizYirik** [comment]
@chaileon Yes, it is in our TODO list for the future releases of MAYGEN but we cant give a  specific date for that new functionality.
